quill-core/src/main/scala/io/getquill/norm/capture/AvoidAliasConflict.scala

```diff
@@ -99,12 +88,50 @@ private case class AvoidAliasConflict(state: collection.Set[Ident])
     else
       loop(x, 1)
   }
+
+  /**
+   * Sometimes we need to change the variables in a function because they will might conflict with some variable
+   * further up in the macro. Right now, this only happens when you do something like this:
+   * <code>
+   * val q = quote { (v: Foo) => query[Foo].insert(v) }
+   * run(q(lift(v)))
+   * </code>
+   * Since 'v' is used by actionMeta in order to map keys to values for insertion, using it as a function argument
+   * messes up the output SQL like so:
+   * <code>
+   *   INSERT INTO MyTestEntity (s,i,l,o) VALUES (s,i,l,o) instead of (?,?,?,?)
+   * </code>
+   * Therefore, we need to have a method to remove such conflicting variables from Function ASTs
+   */
+  private def applyFunction(f: Function): Function = {
+    val (newBody, _, newParams) =
+      f.params.foldLeft((f.body, state, List[Ident]())) {
+        case ((body, state, newParams), param) => {
+          val fresh = freshIdent(param)
+          val pr = BetaReduction(body, param -> fresh)
+          val (prr, t) = AvoidAliasConflict(state + fresh)(pr)
+          (prr, t.state, newParams :+ fresh)
+        }
+      }
+    Function(newParams, newBody)
+  }
 }
 
-private[capture] object AvoidAliasConflict {
+private[getquill] object AvoidAliasConflict {
 
   def apply(q: Query): Query =
     AvoidAliasConflict(collection.Set[Ident]())(q) match {
       case (q, _) => q
     }
+
+  private val dangeriousVars = Set("v").map(Ident(_))
+
+  /**
+   * Make sure query parameters do not collide with paramters of a AST function. Do this
+   * by walkning through the function's subtree and transforming and queries encountered.
+   */
+  def sanitizeVariables(f: Function): Function = {
```

Comment:
`dangeriousVars` would better be a parameter of `sanitizeVariables` I think

Reply:
Good suggestion. I put `dangeriousVars` in `Parsing`.